src/components/legend.ts

```diff
@@ -144,13 +164,17 @@ export module Components {
 
       let availableWidthForEntries = Math.max(0, (availableWidth - this._padding));
 
-      let entryNames = this._colorScale.domain().slice();
-      entryNames.sort(this.comparator());
+      let entryLabels: d3.Map<string> = d3.map<string>();
+      this._colorScale.domain().slice().forEach((d) => {
+        entryLabels.set(this._formatter(d), d);
+      });
+      let entryNames = entryLabels.keys();
+      entryNames = entryNames.sort(this.comparator()).map((d) => entryLabels.get(d));
```

Comment:
>What do you mean by this?

 We use `this._colorScale.scale(domainNames)` to get the color of a entry symbol. And formatter can only translate from domain names to displayed texts. Therefore, I need to store domain names somewhere. That's the reason I want an array of domain names sorted by its corresponding displayed texts instead of an array of sorted displayed texts.